src/App.tsx

```diff
@@ -1,60 +1,143 @@
 import './App.scss';
+import { useEffect, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
 export const App = () => {
+  const [titleInput, setTitleInput] = useState('');
+  const [userSelect, setUserSelect] = useState(0);
+  const [checkInput, setCheckInput] = useState(false);
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserError, setIsUserError] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState([...todosFromServer]);
+
+  useEffect(() => {
+    if (Boolean(titleInput.trim().length) && Boolean(+userSelect)) {
+      setCheckInput(true);
+    }
+
+    // eslint-disable-next-line no-extra-boolean-cast
+    if (Boolean(titleInput.length)) {
+      setIsTitleError(false);
+    }
+
+    // eslint-disable-next-line no-extra-boolean-cast
+    if (Boolean(+userSelect)) {
+      setIsUserError(false);
+    } else {
+      setCheckInput(false);
+    }
+  }, [titleInput, userSelect]);
+
+  const addTodo = () => {
+    if (checkInput && Boolean(titleInput.trim().length)) {
+      const id = Math.max(...visibleTodos.map(todo => todo.id));
+      const newTodo = {
+        id: id + 1,
+        title: titleInput,
+        userId: +userSelect,
+        completed: false,
+      };
+
+      setVisibleTodos([
+        ...visibleTodos, newTodo,
+      ]);
+
+      setTitleInput('');
+      setUserSelect(0);
+    }
+
+    if (titleInput.trim().length === 0) {
+      setIsTitleError(true);
+    }
+
+    if (+userSelect === 0) {
+      setIsUserError(true);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={titleInput}
+            onChange={(event) => {
+              setTitleInput(event.target.value);
```

Comment:
BTW better here use `trim()`

Reply:
I haven't changed it, because it won't let the user use space button at all